src/util/uploads.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * @flow
+ * @file Utility functions for uploads
+ * @author Box
+ */
+
+/**
+ * Returns true if the given object is a Date instance encoding a valid date
+ * (i.e. new Date('this is not a timestamp') should return false).
+ *
+ * Code adapted from
+ * http://stackoverflow.com/questions/1353684/detecting-an-invalid-date-date-instance-in-javascript
+ *
+ * @param {Date} date
+ * @returns {boolean}
+ */
+function isValidDateObject(date: Date) {
+    return Object.prototype.toString.call(date) === '[object Date]' && !isNaN(date.getTime());
+}
+
+/**
+ * Remove milliseconds from date time string
+ * @param {Date} date
+ * @returns {string}
+ */
+function toISOStringNoMS(date: Date) {
+    return date.toISOString().replace(/\.[0-9]{3}/, '');
+}
+
+/**
+ * Returns the file's last modified date as an ISO string with no MS component (e.g.
+ * '2017-04-18T17:14:27Z'), or null if no such date can be extracted from the file object.
+ * (Nothing on the Internet guarantees that the file object has this info.)
+ *
+ * @param {File} file
+ * @returns {?string}
+ */
+function getFileLastModifiedAsISONoMSIfPossible(file: File) {
+    if (isValidDateObject(file.lastModifiedDate)) {
+        return toISOStringNoMS(file.lastModifiedDate);
+    }
+
+    if (
+        file.lastModified &&
+        (typeof file.lastModified === 'string' ||
+            typeof file.lastModified === 'number' ||
+            file.lastModified instanceof Date)
+    ) {
+        const lastModifiedDate = new Date(file.lastModified);
+        if (isValidDateObject(lastModifiedDate)) {
```

Comment:
What do we do for Safari?
https://developer.mozilla.org/en-US/docs/Web/API/File/lastModified

Instead of returning null, can we just return the current Date?

Also, is there a reason to check if its valid? Doesn't the File api guarantee that?

Reply:
1. Not sure about safari, but I tried the example and safari was able to show a valid lastModified.
2. I'm not sure, will discuss with the original author, keeping it as is for now
3. hmm I guess if would be safe to drop the validation

Reply:
actually, not sure about the validation, also keeping it as is and will discuss with the original author (on PTO)